### DISCUSSION TITLE
Why AttestationObject is @Nullable?

### DISCUSSION THREAD

**Discussion by daniel-mezo** [webauthn4j/webauthn4j#644] [Q&A]
Hi guys, @ynojima,

I was wondering why the `AttestationObject` in the `RegistrationRequest` was made `@Nullable` [1]? Isn't that always required to complete registration?
In the spec, I don't see it as optional [2].

[1] https://github.com/webauthn4j/webauthn4j/blob/master/webauthn4j-core%2Fsrc%2Fmain%2Fjava%2Fcom%2Fwebauthn4j%2Fdata%2FRegistrationRequest.java#L40
[2] https://www.w3.org/TR/webauthn-2/#dom-authenticatorattestationresponse-attestationobject

Cheers,
Daniel

**Comment by daniel-mezo** [comment]
It is also nullable in the [RegistrationData](https://github.com/webauthn4j/webauthn4j/blob/master/webauthn4j-core/src/main/java/com/webauthn4j/data/RegistrationData.java#L39) btw.

**Comment by ynojima** [comment]
`RegistrationRequest`, `RegistrationData`, `AuthenticationRequest`, and `AuthenticationData` fields are all `nullable` because they represents untrusted inputs from a client.

**Reply by daniel-mezo** [reply]
But how could the ceremony continue without these inputs? I would think if a data point is required to continue the operation, it should be present, i.e. not null, no?

**Reply by ynojima** [reply]
You are right. it cant be continued without that. It will end with error.
I thought these class should have ability to represent broken data even if the ceremony cannot continue.
